src/parser/compose.rs

```diff
@@ -318,402 +224,40 @@ impl ComposeParser {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn parse_with_env(content: &str, env_path: &Path) -> Result<ComposeConfig, DockerError> {
-        // Read environment variables from file
-        let env_content = std::fs::read_to_string(env_path)
-            .map_err(|e| DockerError::ValidationError(format!("Failed to read env file: {}", e)))?;
-
-        // Parse environment variables using existing function
-        let env_vars = Self::parse_env_file(&env_content)?;
-
-        // Substitute environment variables in the content
-        let content = Self::substitute_env_vars(content, &env_vars)?;
-
-        // Parse the content with substituted environment variables
-        Self::parse(&content)
+    pub fn env_vars(mut self, vars: HashMap<String, String>) -> Self {
+        self.env_vars = Some(vars);
+        self
     }
+}
 
-    /// Substitutes environment variables in a string
-    ///
-    /// Supports the following formats:
-    /// - ${VAR}
-    /// - ${VAR:-default}
-    /// - $VAR
-    ///
-    /// # Arguments
-    ///
-    /// * `content` - The string containing environment variable references
-    /// * `env_vars` - HashMap of environment variables to use for substitution
-    fn substitute_env_vars(
```

Comment:
Why get rid of this?

Reply:
It still exists, it's just in this file https://github.com/tangle-network/dockworker/blob/serial/cleanup/src/parser/env.rs